src/components/Permalink/Permalink.js

```diff
@@ -17,24 +15,139 @@ const propTypes = {
   history: PropTypes.shape({
     replace: PropTypes.func,
   }),
+
+  /**
+   * Layers provider.
+   */
+  layerService: PropTypes.instanceOf(LayerService),
+
+  /**
+   * An ol map.
+   */
+  map: PropTypes.instanceOf(OLMap),
+
+  /**
+   * Params to be written in url.
+   */
+  params: PropTypes.object,
 };
 
 const defaultProps = {
   history: null,
+  layerService: null,
+  map: null,
+  params: {},
 };
 
 /**
  * This component handles permalink logic.
  */
 class Permalink extends Component {
-  componentDidUpdate() {
+  constructor(props) {
+    super(props);
+    this.state = {
+      x: undefined,
+      y: undefined,
+      z: undefined,
+      layers: undefined,
+    };
+    this.moveEndRef = null;
+  }
+
+  componentDidMount() {
+    const { map, layerService } = this.props;
+    if (map) {
+      this.moveEndRef = map.on('moveend', () => {
+        this.onMapMoved();
+      });
+    }
+
+    if (layerService) {
+      this.updateLayerService();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { map, layerService } = this.props;
+
+    if (layerService !== prevProps.layerService) {
+      this.updateLayerService();
+    }
+
+    if (map !== prevProps.map) {
+      map.on('moveend', () => this.onMapMoved());
```

Comment:
you forget to reset the ref
```suggestion
      this.moveEndRef = map.on('moveend', () => this.onMapMoved());
```